scripts/qa_crowbarsetup.sh

```diff
@@ -3860,6 +3908,98 @@ function oncontroller_heat_image_setup()
     fi
 }
 
+function oncompute_parse_vm_xml
+{
+    local instance_name=$1
+    echo $instance_name
+    virsh list --all | grep $instance_name
+    [ $? != 0 ] && complain 53 "failed to find pci-passthru-instance"
+
+    # verify node xml to see if it has pci passthru section which has the
+    # specific address passing from compute host
+    virsh dumpxml $instance_name | grep -E "<hostdev" -A 7 | grep "0x1d"
+    [ $? != 0 ] && complain 63 "failed to find pci-passthru block in node xml"
+
+    return 0
+}
+
+function oncontroller_retrieve_image
+{
+    local cirros_image_name=$1
```

Comment:
The name of this function is generic but then the variable naming assumes it's only for cirros.  I suggest you drop the `cirros_` prefix so it can be reused by anything else which might want to download images.